File: android/Book-AndroidProgramming/CriminalIntentJ/app/src/main/java/kr/ac/ajou/heidi/criminalintentj/database/CrimeCursorWrapper.java
package kr.ac.ajou.heidi.criminalintentj.database;

import android.database.Cursor;
import android.database.CursorWrapper;

import java.util.Date;
import java.util.UUID;

import kr.ac.ajou.heidi.criminalintentj.database.CrimeDbSchema.CrimeTable;
import kr.ac.ajou.heidi.criminalintentj.model.Crime;


public class CrimeCursorWrapper extends CursorWrapper {

    public CrimeCursorWrapper(Cursor cursor) {
        super(cursor);
    }

    public Crime getCrime() {
        String uuidString = getString(getColumnIndex(CrimeTable.Cols.UUID));
        String title = getString(getColumnIndex(CrimeTable.Cols.TITLE));
        long date = getLong(getColumnIndex(CrimeTable.Cols.DATE));
        int isSolved = getInt(getColumnIndex(CrimeTable.Cols.SOLVED));
        String suspect = getString(getColumnIndex(CrimeTable.Cols.SUSPECT));

        Crime crime = new Crime(UUID.fromString(uuidString));
        crime.setmTitle(title);
        crime.setmDate(new Date(date));
        crime.setmSolved(isSolved != 0);
        crime.setSuspect(suspect);

        return crime;
    }
}

File: android/Book-AndroidProgramming/CriminalIntentK/app/src/main/java/kr/ac/ajou/heidi/criminalintentk/database/CrimeCursorWrapper.kt
package kr.ac.ajou.heidi.criminalintentk.database

import android.database.Cursor
import android.database.CursorWrapper
import kr.ac.ajou.heidi.criminalintentk.model.Crime
import java.util.*

class CrimeCursorWrapper(cursor: Cursor): CursorWrapper(cursor) {

    fun getCrime(): Crime? {
        val uuidString = getString(getColumnIndex(CrimeDbSchema.CrimeTable.Cols.UUID))
        val title = getString(getColumnIndex(CrimeDbSchema.CrimeTable.Cols.TITLE))
        val date = getLong(getColumnIndex(CrimeDbSchema.CrimeTable.Cols.DATE))
        val isSolved = getInt(getColumnIndex(CrimeDbSchema.CrimeTable.Cols.SOLVED))

        val crime = Crime(UUID.fromString(uuidString))
        crime.title = title
        crime.date = Date(date)
        crime.solved = isSolved != 0

        return crime
    }
}